File: Práctica3/lib/ShieldBooster.rb
#encoding: utf-8

require_relative 'ShieldToUI'

# Representa a los potenciadores de escudos que pueden tener
# las naves espaciales

module Deepspace
  class ShieldBooster
    def initialize(a_name, a_boost, a_uses)
      @name = a_name
      @boost = a_boost
      @uses = a_uses
    end

    def self.newCopy(shield)
      new(shield.name, shield.boost, shield.uses)
    end

    def name
      @name
    end

    def boost
      @boost
    end

    def uses
      @uses
    end

    def useIt
      if uses > 0
        @uses -= 1
        @boost
      else
        1.0
      end
    end

    # String representation of the object
    # @return [String] string representation
    def to_s
      message = "[ShieldBooster]-> Boost: #{@boost}, Uses: #{@uses}"
      return message
    end

    # To UI
    def getUIversion
      return ShieldToUI.new(self)
    end
  end
end

# Código de prueba
# prueba =  Deepspace::ShieldBooster.new('prueba', 1, 1)
# puts prueba.name
# prueba2 = Deepspace::ShieldBooster.new('prueba2', 1, 0)
# puts prueba.uses
# pruebacopy = Deepspace::ShieldBooster.new('pruebacopy', 2, 2)
# prueba3 = Deepspace::ShieldBooster.newCopy(pruebacopy)
# puts prueba3.boost

File: Práctica4/java/src/deepspace/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author daniel
 */

// Representa a los potenciadores de escudo de las estaciones espaciales

public class ShieldBooster {
    private String name;
    private float boost;
    private int uses;

    public ShieldBooster(String nam, float bo, int us){
        this.name = nam;
        this.boost = bo;
        this.uses = us;
    }
    
    public ShieldBooster(ShieldBooster s){
        this.name = s.name;
        this.boost = s.boost;
        this.uses = s.uses;
    }
    
    float getBoost(){
        return boost;
    }
    
    int getUses(){
        return uses;
    }
    
    float useIt(){
        if(uses > 0){
            uses-=1;
            return boost;
        }else{
            return 1.0f;
        }
    }
    
    // -------------------------------------------------------------------------
    // String representation, UI version
    // -------------------------------------------------------------------------
    
    /**
     * String representation of the object.
     * @return string representation
     */
    public String toString() {
        String message = "[ShieldBooster] -> Boost: " + boost
                + ", Uses: " + uses;
        return message;
    }
    
    /**
     * To UI.
     */
    ShieldToUI getUIversion() {
        return new ShieldToUI(this);
    }
 }
